modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java

```diff
@@ -639,6 +653,37 @@ private boolean initme() {
         return true;
     }
 
+    /**
+     * Use secure vault to secure password in JMS Message Store
+     * @param newParamValue
+     * @return
+     */
+    private String resolveSecureVaultExpressions(String newParamValue) {
+        Pattern vaultLookupPattern = Pattern.compile(secureVaultRegex);
+        Matcher lookupMatcher = vaultLookupPattern.matcher(newParamValue);
+        //setting value initially
+        if(lookupMatcher.find()) {
+            Value expression = null;
+            //getting the expression with out curly brackets
+            String expressionStr = lookupMatcher.group(1);
+            try {
+                expression = new Value(new SynapseXPath(expressionStr));
+            } catch (JaxenException e) {
+                log.error("Error while building the expression : " + expressionStr);
+            }
+            if (expression != null) {
```

Comment:
null check is not needed if we throw a synapse exception within the catch block. I think its better not to move the code after line 674 into the try block.